### ISSUE TITLE
Provide a new data type which merge all feed types into a single type

### ISSUE THREAD

**Issue by MikaelPorttila** [MikaelPorttila/rss#27]
The idea is to provide a new data type that manage to map all feed types (ATOM,  RSS2, JSONFeed) into a single type.

The work that needs to be done is:

1. List all the needed fields both on feed and entry level.
2. Mapping lists between the new type and all supported feed types. (Suggested mappings could be added to this issue)
3. Build mappers and tests for all feed types.
4. (optional) The idea from the beginning was to use the JsonFeed type as a type which all feeds could map into but there are limitations to the JsonFeed type so once the new mapper is in place the project should replace the outputJsonFeed option with a new transformation option instead. (This will break the API backward compatibility and a major version bump is required)

Atom (https://datatracker.ietf.org/doc/html/rfc5023) 
RSS 2 (https://www.rssboard.org/rss-specification)
RSS 2 (Dublin Core namespace https://www.feedforall.com/dublin-core.htm)
JSON Feed (https://jsonfeed.org/version/1.1)

New type (WORK IN PROGRESS 🛠, latest suggestion based on comments below):
```
{
    title: {
        value: string 
            Atom: atom.title.value
            RSS2: channel.title
            Dublin Core: dc:title
            JSONFeed: title
        type: string
            Atom: atom.title.type
            RSS2: N/A
            Dublin Core: N/A
            JSONFeed: N/A
    }
}
```

**Comment by NfNitLoop** [comment]
I'm building a tool based on your code. Thanks for this library! :) 

As an example case where JSONFeed is insufficient for my needs: it has a `date_published`, but since that's a JavaScript `Date`, it loses the time zone offset of the original timestamp in RSS, so I can't properly translate that to my output format.

**Comment by MikaelPorttila** [comment]
@NfNitLoop I have extended the API in the new release (0.4.0) - you can now access the original date string through the new fields which are suffixed with "Raw". Happy hacking!

**Comment by MikaelPorttila** [comment]
Release 0.5.0 now provide a new parseFeed function which all feed types are merge into
